src/storage/mod.rs

```diff
@@ -742,8 +743,7 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
         start_key: Key,
         end_key: Key,
         notify_only: bool,
-        callback: Callback<()>,
-    ) -> Result<()> {
+    ) -> Result<impl Future<Output = Result<()>>> {
```

Comment:
Can you make it an async function instead of using an implicit future?

Reply:
Sorry it's from another branch, I made a rebase mistake ……